rvps/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#![allow(clippy::new_without_default)]
+
+pub mod cache;
+pub mod extractors;
+pub mod pre_processor;
+pub mod reference_value;
+
+use std::time::SystemTime;
+
+use anyhow::{anyhow, Result};
+use chrono::{DateTime, Utc};
+use extractors::{Extractors, ExtractorsImpl};
+use pre_processor::{PreProcessor, PreProcessorAPI, Ware};
+use serde::{Deserialize, Serialize};
+
+pub use cache::Cache;
+pub use reference_value::{ReferenceValue, TrustedDigest};
+
+/// Default version of Message
+static MESSAGE_VERSION: &str = "0.1.0";
+
+/// Message is an overall packet that Reference Value Provider Service
+/// receives. It will contain payload (content of different provenance,
+/// JSON format), provenance type (indicates the type of the payload)
+/// and a version number (use to distinguish different version of
+/// message, for extendability).
+/// * `version`: version of this message.
+/// * `payload`: content of the provenance, JSON encoded.
+/// * `type`: provenance type of the payload.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Message {
+    #[serde(default = "default_version")]
+    version: String,
+    payload: String,
+    r#type: String,
```

Comment:
For `type` is a key word in Rust. If we use another name like `typ`, there should be a `#[serde(rename = "type")]` macro. IMO, it might not be a big problem, for different habits for persons.